### PR TITLE
Update Sonarr/Radarr to use v3 api

### PR DESCRIPTION
**Pull request by stevezau**
### Motivation for changes:

The latest versions of Sonarr and Radarr have removed the legacy API. We need to start using API v3. I believe it is as simple as changing the URL path. 

I've changed this on my system and ill test for a few days.

#### To Do:

- [ ] Do we need to handle old versions of Sonarr/Radarr??

### CODE REVIEW

**Comment by paranoidi** [comment]
Good question about backward compatibility, we have a similar standing issue with #2906

I think we need some sort of of (soft) policy. It's always annoying if updating software x then requires upgrading bunch of other applications too, but we don't have too many maintainers / testers to keep everything tested to be backwards compatible ....

I'd say we note these upgrade requirements in changelog and support only latest stable releases.

@gazpachoking, any opinions?

**Comment by stevezau** [comment]
@paranoidi i believe Radarr and Sonarr v3 and v4 (latest unstable) both support v3 API. So this may be a non-issue.

**Comment by stevezau** [comment]
I've been running off this brach for 24 hours, so far no errors.

**Comment by stevezau** [comment]
@gazpachoking @paranoidi been running for a while now. It seems stable.

Thoughts on how we handle legacy? 

NOTE: I do need to look into the current versions of sonarr/radarr to understand what API level they support vs the bleeding edge.

**Comment by stevezau** [comment]
@gazpachoking @paranoidi thoughts on merge?

**Comment by Dracon09** [comment]
Will this be included into the main branch soon?

File: flexget/components/imdb/api.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
+from __future__ import absolute_import, division, unicode_literals
```
Comment (gazpachoking) [inline flexget/components/imdb/api.py:1]: something go wrong and this file got reverted to an old version? Seems like utils/imdb.py showed back up again too

File: flexget/tests/test_metainfo.py [hunk 2]
```diff
@@ -1,7 +1,11 @@
+from __future__ import absolute_import, division, unicode_literals
+
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
```
Comment (gazpachoking) [inline flexget/tests/test_metainfo.py:3]: This file get reverted during a merge conflict too?

**Comment by patrickegli** [comment]
@stevezau Can you review #3771 and accept the MR, so we can move on with this MR enabling support for radarr/sonarr v3 api? Thx.

**Comment by stevezau** [comment]
@gazpachoking @paranoidi this has been working for me for a long time.. Should we merge?

It only handles v3 API from sonarr/radarr.. Given they have been out for a while maybe that's ok?

**Comment by stevezau** [comment]
> @stevezau Can you review #3771 and accept the MR, so we can move on with this MR enabling support for radarr/sonarr v3 api? Thx.

looks like there might be 1 or 2 more merge conflicts?

**Comment by gazpachoking** [comment]
The commit list is suspiciously long, and the actual changes are suspiciously short. Are those few changes the only things actually needed? If so it shouldn't be too hard to fix up the conflicts.

**Comment by gazpachoking** [comment]
> It only handles v3 API from sonarr/radarr.. Given they have been out for a while maybe that's ok?

I'm fine with it.

**Comment by stevezau** [comment]
> The commit list is suspiciously long, and the actual changes are suspiciously short. Are those few changes the only things actually needed? If so it shouldn't be too hard to fix up the conflicts.

I think it is because I've kept it up to date with develop, at one point. I also opened it a long time ago. But yes, the file changes as fairly small.

**Comment by patrickegli** [comment]
> > @stevezau Can you review #3771 and accept the MR, so we can move on with this MR enabling support for radarr/sonarr v3 api? Thx.
> 
> looks like there might be 1 or 2 more merge conflicts?

Merged branch develop and fixed conflicts: #3775

**Comment by stevezau** [comment]
@patrickegli odd it is still showing conflicts?

**Comment by stevezau** [comment]
@patrickegli i merged in your changes but it caused a whole lot more merge conflicts.. I didn't review your changes as I don't have time.. 

I am not sure what has happened?

**Comment by patrickegli** [comment]
> @patrickegli i merged in your changes but it caused a whole lot more merge conflicts.. I didn't review your changes as I don't have time..
> 
> I am not sure what has happened?

I don't know exactly what went wrong, because all checks passed in my MR. Maybe it's because i merged branch develop in my fork repository and not in original repo? 🤔 
Easiest would be if you can grant me write access on this branch?

**Comment by stevezau** [comment]
@patrickegli i added you to the team so you should have access

**Comment by patrickegli** [comment]
> @patrickegli i added you to the team so you should have access

```
06:50:18.295: [Flexget] git push
remote: Permission to Flexget/Flexget.git denied to patrickegli.
fatal: unable to access 'https://github.com/Flexget/Flexget.git/': The requested URL returned error: 403
```

Doesn't work yet... Other option would be to create a new MR with a clean start from current develop instead of this almost two years old branch. What do you think?

**Comment by patrickegli** [comment]
> > @patrickegli i added you to the team so you should have access
> Doesn't work yet... Other option would be to create a new MR with a clean start from current develop instead of this almost two years old branch. What do you think?

Sorry, had to join the team first... Works now.

**Comment by stevezau** [comment]
All tests are passing. Merging it in.

**Pull request merged**